vllm/distributed/device_communicators/custom_all_reduce.py

```diff
@@ -138,6 +138,12 @@ def __init__(self,
         else:
             device_ids = list(range(torch.cuda.device_count()))
 
+        if len(device_ids) < world_size:
+            logger.warning(
+                "Custom allreduce is disabled because this feature is "
+                "not intended for multi node use case.")
+            return
+
```

Comment:
This might be wrong. We cannot assume `CUDA_VISIBLE_DEVICES` is always set. If not set, `device_ids` will be all devices (e.g. 8), but we should be able to use custom allreduce with `world_size` 2 or 4.